### PR TITLE
refactor of frame filters & support pairs for serviceNames and responses

### PR DESCRIPTION
**Pull request by ShanniLi**
r @kriskowal @Raynos 
cc @jcorbin 

1. update hexer dependency
2. fixed a small bug in coloring
3. refactor of frame filters: all filters are put into one file.
4. call res is displayed when filtered by service names.
5. call req is displayed when filtered by response statuses.

### CODE REVIEW

**Comment by Raynos** [comment]
Seems reasonable to me; lgtm.

File: bin/tcap.js [hunk 1]
```diff
@@ -101,6 +100,20 @@ function checkUid() {
...
+function registerFilters() {
+    var filters = new FrameFilters();
+
+    // this order of registration matters
+    filters.register('serviceName',
+        commander.service.length ? commander.service : null);
+    filters.register('arg1',
+        commander.arg1.length ? commander.arg1 : null);
+    filters.register('response',
+        checkResponse(commander.response));
```
Comment (kriskowal) [inline bin/tcap.js]: I would avoid taking advantage of the `commander` module’s shared global state and pass it around as if it were an instance.

File: frame-filters.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    if ((frame.body.type === FrameTypes.CallResponse) ||
+        (frame.body.type === FrameTypes.CallResponseCont && !frame.flag)) {
+        return true;
+    }
+
+    return false;
+}
+
+//
+// the fitler on service name
```
Comment (kriskowal) [inline frame-filters.js]: s/fitler/filter

File: frame-filters.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        return false;
+    }
+
+    ids[frame.id] = true;
+
+    return true;
+};
+
+//
+// the fitler on arg1
```
Comment (kriskowal) [inline frame-filters.js]: s/fitler

File: frame-filters.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        return false;
+    }
+
+    ids[frame.id] = true;
+
+    return true;
+};
+
+//
+// the fitler on responses
```
Comment (kriskowal) [inline frame-filters.js]: s/fitler

**Pull request merged**